script.js

```diff
@@ -185,222 +181,142 @@ const books = [
-      '2 tablespoons tomato paste',
-      '5 cloves garlic',
-      '2 bay leaves',
-      '3 sprigs thyme',
-      '3 sprigs Italian parsley',
-      '1/2 teaspoon black peppercorns'
-    ],
-    source: 'Food52',
-    totalTime: 60,
-    url: 'https://food52.com/recipes/3712-meat-stock',
-    image: './recipe-images/meat.jpg'
-  },
-  {
-    name: 'Homemade Meat Broth',
-    cuisineType: 'american',
-    ingredients: [
-      '1 teaspoon salt',
-      '1 carrot, peeled',
-      '1 medium onion, peeled',
-      '2 or 3 celery stalks',
-      '¼ red or yellow bell pepper, stripped of all its seeds',
-      '1 small boiling potato, peeled',
-      '1 ripe, fresh tomato, or 1 canned Italian plum tomato, drained of juice',
-      '5 pounds assorted pieces of meat and bones (see meat suggestions above), of which no less than 1½ pounds is all meat'
-    ],
-    source: 'Cookstr',
-    totalTime: 60,
-    url: 'http://www.cookstr.com/recipes/il-brodo-homemade-meat-broth',
-    image: './recipe-images/meat.jpg'
-  },
-  {
-    name: 'Spice-Rubbed Grilled Flap Meat (Sirloin Tip) Recipe',
-    cuisineType: 'south-american',
-    ingredients: [
-      '1 tablespoon whole black peppercorns, toasted',
-      '1 teaspoon coriander seed, toasted',
-      '1 teaspoon fennel seed, toasted',
-      '1 teaspoon cumin pods, toasted',
-      '1 teaspoon red pepper flakes',
-      '1/2 teaspoon dried oregano',
-      '2 medium cloves garlic, minced (about 2 teaspoons)',
-      '2 tablespoons vegetable or canola oil',
-      '1 whole flap meat steak, 2 to 2 1/2 pounds',
-      'Kosher salt'
-    ],
-    source: 'Serious Eats',
-    totalTime: 240,
-    url: 'http://www.seriouseats.com/recipes/2012/05/spice-rubbed-grilled-flap-meat-sirloin-tip-recipe.html',
-    image: './recipe-images/grilled.jpg'
-  }
-]
+//DOM
+const fictionButton = document.getElementById('filterFiction')
+const scienceButton = document.getElementById('filterScienceFiction')
+const fantasyButton = document.getElementById('filterFantasy')
+const adventureButton = document.getElementById('filterAdventure')
+const horrorButton = document.getElementById('filterHorror')
+const mysteryButton = document.getElementById('filterMystery')
+const dystopianButton = document.getElementById('filterDystopian')
+const ascendingButton = document.getElementById('ascending')
+const descendingButton = document.getElementById('descending')
+const filterBar = document.getElementById('filterBar')
+const rateButton = document.getElementById('from-high-rate')
+const randomDiv = document.getElementById('random-book-div')
+const randomButton=document.getElementById('random-button')
+
+//creat element
+const galleryWrapperDiv = document.createElement('div')
+
+//genre type by filter for fiction
+const genreFiction = () => {
+  const filteredFiction = books.filter((book)=>book.genre === 'Fiction')
+displayBooks(filteredFiction)
+}
+fictionButton.addEventListener('click',genreFiction)
```

Comment:
write a generic function instead, using the genre as parameter to check, look at this code:

```
const filterAndDisplayBooksByGenre = (genre) => {
  const filteredBooks = books.filter(book => book.genre === genre);
  displayBooks(filteredBooks);
}

fictionButton.addEventListener('click', () => filterAndDisplayBooksByGenre('Fiction'));
```